redaxo/src/addons/structure/lib/service_slice.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/**
+ * @package redaxo\structure
+ */
+class rex_slice_service
+{
+    /**
+     * @param array $data
+     *
+     * @throws \rex_api_exception
+     * @throws \rex_sql_exception
+     *
+     * @return string
+     */
+    public static function addSlice($data): string
+    {
+        if (!is_array($data)) {
+            throw new rex_api_exception('Expecting $data to be an array!');
+        }
+        self::reqKey($data, 'clang_id');
+        self::reqKey($data, 'ctype_id');
+        self::reqKey($data, 'article_id');
+        self::reqKey($data, 'module_id');
+        self::reqKey($data, 'priority');
+        if (rex_plugin::get('bloecks', 'status')->isAvailable()) {
+            self::reqKey($data, 'status');
+        }
+        if (rex_plugin::get('structure', 'version')->isAvailable()) {
+            self::reqKey($data, 'revision');
+        }
+
+        if ($data['priority'] <= 0) {
+            $data['priority'] = 1;
+        }
+
+        $message = rex_i18n::msg('slice_added');
+
+        $ASLICE = rex_sql::factory();
+        $user = self::getUser();
+
+        $ASLICE->setTable(rex::getTablePrefix() . 'article_slice');
+        $slice_id = $ASLICE->setNewId('id');
```

Comment:
`setNewId` braucht man nur bei den IDs, die nicht Primary Key sind. In rex_article_slice ist id aber PK, somit kann man sich auf den auto_increment-Wert verlassen.

Reply:
Ah ok. Nehm ich raus.